### PR TITLE
Feat/action

### PR DESCRIPTION
**Pull request by giovannic**
I have a bunch of repositories, and counting, that I want to set up pkgdown <-> gh-pages for. This seems like the easiest way to do it.

What do you think?

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+git checkout "$GITHUB_SHA"
+
+# set up docs to work on the gh-branch
+git worktree add $DEST gh-pages
+
+echo "#################################################"
+echo "Building the documentation"
+echo "#################################################"
+
+R -e "library(pkgdown); build_site('.', \
```
Comment (richfitz) [inline entrypoint.sh]: How does this cope with other dependent packages required to run the pkgdown site? Do these need to be installed into the docker container?

**Comment by giovannic** [comment]
> 
> 
> Have you seen this? https://github.com/r-lib/actions/blob/c4ec589084d971218fd6f4242b0265494dfbb368/examples/pkgdown.yaml
> 
> I've not used actions and don't know how they actually work, so might be being daft, but does this mean it's another thing that will have to install all the dependencies, and we can't use the copy of dependencies that travis will have already installed? Can we do something like build the pkgdown in the travis task and then have this action just take care of pushing that onto the github branch?

Oh it already exists! And it came out yesterday!

The appeal of actions for me is that they are easier to set up and reusable. With Travis you have to set up deploy keys and use this shaky  `deploy_site_github` code (or create your own script to copy from repo to repo). It doesn't seem to run on R 3.6.1 (because the Travis package isn't yet available) and needs a lot of jiggling to set up. Whereas now you only need to add a config file.

@richfitz The r-lib action looks like it could work. Would you prefer that? Or to stay with Travis?

File: entrypoint.sh (hunk 1 above)
Comment (giovannic) [reply entrypoint.sh]: The dependencies are installed in the [Dockerfile](https://github.com/mrc-ide/pkgdown-action/pull/1/files#diff-3254677a7917c6c01f55212f86c57fbf)

**Comment by giovannic** [comment]
After digging into the r-lib issues, it's quite clear that this will come with other [cons](https://github.com/maxheld83/ghpages/issues/1). I will park this and live with the janky setup for now.

**Pull request merged**